src/Extensions/SiteTreeSubsites.php

```diff
@@ -412,9 +412,7 @@ public function alternateAbsoluteLink($action = null)
         // Generate the existing absolute URL and replace the domain with the subsite domain.
         // This helps deal with Link() returning an absolute URL.
         $url = Director::absoluteURL($this->owner->Link($action));
-        if ($this->owner->SubsiteID) {
-            $url = preg_replace('/\/\/[^\/]+\//', '//' . $this->owner->Subsite()->domain() . '/', $url);
-        }
+        $url = preg_replace('/\/\/[^\/]+\//', '//' . $this->owner->Subsite()->domain() . '/', $url);
```

Comment:
Why are we removing this check? If $this->owner->SubsiteID == 0 then this will cause an exception i.e. `(null)->domain()`

Reply:
I'm about 60% sure `$this->owner->Subsite()` would give a new Subsite object with no ID - but that would need to be checked if someone picks this PR up again. If that's not the case, then another mechanism will need to be added here to get the main site Domain where no subsite is set.

Reply:
I'm 100% sure.
https://github.com/silverstripe/silverstripe-framework/blob/4/src/ORM/DataObject.php#L1890